### ISSUE TITLE
Milvus's insert was blocked in multiprocessing with python

### ISSUE THREAD

**Issue by HuangdaxianZaixian** [milvus-io/milvus#3172]
**Describe the bug**
I created a multiprocessing procedure with python, and inserted feature vectors to milvus at each process, however the procedure was blocked at the insert action.

**Steps/Code to reproduce behavior**

from milvus import *
from multiprocessing import Manager, Process
from time import sleep
import numpy as np
from multiprocessing import Process

MILVUS_SERVER_IP = "192.168.1.147"
MILVUS_SERVER_PORT = "19530"

class MilvusClient:

    def __init__(self, serverIP, serverPort):
        self._milvusConn = self.connectMilvus(serverIP, serverPort)

    @staticmethod
    def connectMilvus(serverIP, serverPort):
        __conn__  = Milvus(serverIP, serverPort)
        if not __conn__:
            print("failed to connect to milvus server with ip {0} and port {1}".format(serverIP, serverPort))
            exit(-1)
        return __conn__

    def __addMilvusCollectionIndex(self, collectionName):
        index_param = {'nlist': 16384}
        status = self._milvusConn.create_index(collectionName, IndexType.IVFLAT, index_param)
        if not status.OK():
            print("failed to create collection index with status: {}".format(status))
            print("status: ", status)
            exit(-1)

    def createMilvusCollection(self, collectionName):

        if not self._milvusConn.has_collection(collectionName)[1]:

            param = {
                'collection_name': collectionName,
                'dimension': 2560,
                'index_file_size': 1024,
                'metric_type': MetricType.IP}
            self._milvusConn.create_collection(param)

            self.__addMilvusCollectionIndex(collectionName)

        else:
            print("collection {0} has already exist".format(collectionName))

    def addFeatureVectors(self, collectionName, featureVectors, workerID):
        print("workerID {}: addFeatureVectors 11".format(workerID))
        status, ids = self._milvusConn.insert(collection_name = collectionName, records = featureVectors)
        print("workerID {}: addFeatureVectors 22".format(workerID))
        if not status.OK():
            print("failed to insert vector to collection {0}".format(collectionName))
            print("status: ", status)
            exit(-1)

        print("workerID {}: addFeatureVectors 33".format(workerID))

        return ids

    @staticmethod
    def loadFeatureVector(vectorPath):
        data = []
        with open(vectorPath, "r") as bfile:
            data = np.fromfile(bfile, dtype=np.uint8)
            data = (data + 0.5) / 255

        return np.array([data])
    
class worker(Process):

    def __init__(self, _workerID, messageQueue):

        super(worker, self).__init__()
        
        self.workerID = _workerID
        self._messageQueue = messageQueue

        self.milvusClient = MilvusClient(MILVUS_SERVER_IP, MILVUS_SERVER_PORT)
        self.collectionName = "PicFeture"
        self.milvusClient.createMilvusCollection(self.collectionName)
        

    def run(self):

        print("workerID {0} start".format(self.workerID))

        while True:
            try:
                temp = self._messageQueue.get_nowait()

                featureVectors = self.milvusClient.loadFeatureVector(temp)
                print("workerID {0} running 0".format(self.workerID))
                self.milvusClient.addFeatureVectors(self.collectionName, featureVectors, self.workerID)
                print("workerID {0} running 1".format(self.workerID))

            except Exception as err:
                print("workerID {0} exception: {1}".format(self.workerID, err))
                sleep(2)

        print("workerID {0} exit".format(self.workerID))

def runWorkers(messageQueue, workerNum = 4):

    processList = []
    for processIndex in range(workerNum):
        _process = worker(processIndex, messageQueue)
        processList.append(_process)

    for _process in processList:
        _process.start()

if __name__ == "__main__":

    _manager = Manager()
    messageQueue = _manager.Queue()

    runWorkers(messageQueue)
    
    while True:
        picUrl = "./feature.bin"
        messageQueue.put(picUrl)

        sleep(1)

**Expected behavior**
workerID 0 start
workerID 0 exception: 
workerID 1 start
workerID 2 start
workerID 3 start
workerID 2 exception: 
workerID 1 running 0
workerID 1: addFeatureVectors 11
workerID 3 exception: 
workerID 0 running 0
workerID 0: addFeatureVectors 11
workerID 3 exception: 
workerID 2 running 0
workerID 2: addFeatureVectors 11
workerID 3 running 0
workerID 3: addFeatureVectors 1

**Environment details**
- Hardware/Softward conditions: ubuntu 16.04
- Method of installation:  docker, python3.6
- Milvus version: milvus_cpu_0.10.0
- Milvus configuration: default

**Comment by BossZou** [comment]
@HuangdaxianZaixian May you provide the server log?

**Comment by BossZou** [comment]
@HuangdaxianZaixian , I tried your procedure in my host, and it output :
```
workerID 2 exception: [Errno 2] No such file or directory: './feature.bin'
workerID 0 exception: [Errno 2] No such file or directory: './feature.bin'
```
So, which assets I need to make sure the procedure run correctly?

**Comment by HuangdaxianZaixian** [comment]
> @HuangdaxianZaixian , I tried your procedure in my host, and it output :
> 
> ```
> workerID 2 exception: [Errno 2] No such file or directory: './feature.bin'
> workerID 0 exception: [Errno 2] No such file or directory: './feature.bin'
> ```
> 
> So, which assets I need to make sure the procedure run correctly?

"./feature.bin" is just a feature vector file for test
![feature](https://user-images.githubusercontent.com/23063959/89844592-99d6b900-dbae-11ea-8d05-22c663263e72.jpg)

**Comment by HuangdaxianZaixian** [comment]
> @HuangdaxianZaixian May you provide the server log?

[milvus-20-08-12-13:45-info.log](https://github.com/milvus-io/milvus/files/5054185/milvus-20-08-12-13.45-info.log)

**Comment by BossZou** [comment]
@HuangdaxianZaixian , I run this procedure and set `workerNum` to 1, it was blocked still. And I tried to modify class 'worker' as follow:
```
class worker(Process):
    def __init__(self, _workerID, messageQueue):

        super(worker, self).__init__()

        self.workerID = _workerID
        self._messageQueue = messageQueue

        self.milvusClient = None
        self.collectionName = "PicFeture"

    def run(self):
        self.milvusClient = MilvusClient(MILVUS_SERVER_IP, MILVUS_SERVER_PORT)
        self.milvusClient.createMilvusCollection(self.collectionName)

        print("workerID {0} start".format(self.workerID))

        while True:
            try:
                temp = self._messageQueue.get_nowait()

                featureVectors = self.milvusClient.loadFeatureVector(temp)
                print("workerID {0} running 0".format(self.workerID))
                self.milvusClient.addFeatureVectors(self.collectionName, featureVectors, self.workerID)
                print("workerID {0} running 1".format(self.workerID))

            except Exception as err:
                print("workerID {0} exception: {1}".format(self.workerID, err))
                sleep(2)

        print("workerID {0} exit".format(self.workerID))

```

As you see above, I create a milvus client in member function `run ` rather than `__init__`. It take effect.

**Comment by HuangdaxianZaixian** [comment]
> @HuangdaxianZaixian , I run this procedure and set `workerNum` to 1, it was blocked still. And I tried to modify class 'worker' as follow:
> 
> ```
> class worker(Process):
>     def __init__(self, _workerID, messageQueue):
> 
>         super(worker, self).__init__()
> 
>         self.workerID = _workerID
>         self._messageQueue = messageQueue
> 
>         self.milvusClient = None
>         self.collectionName = "PicFeture"
> 
>     def run(self):
>         self.milvusClient = MilvusClient(MILVUS_SERVER_IP, MILVUS_SERVER_PORT)
>         self.milvusClient.createMilvusCollection(self.collectionName)
> 
>         print("workerID {0} start".format(self.workerID))
> 
>         while True:
>             try:
>                 temp = self._messageQueue.get_nowait()
> 
>                 featureVectors = self.milvusClient.loadFeatureVector(temp)
>                 print("workerID {0} running 0".format(self.workerID))
>                 self.milvusClient.addFeatureVectors(self.collectionName, featureVectors, self.workerID)
>                 print("workerID {0} running 1".format(self.workerID))
> 
>             except Exception as err:
>                 print("workerID {0} exception: {1}".format(self.workerID, err))
>                 sleep(2)
> 
>         print("workerID {0} exit".format(self.workerID))
> ```
> 
> As you see above, I create a milvus client in member function `run ` rather than `__init__`. It toke effect.

Yes, it was reported in https://github.com/milvus-io/milvus/issues/3032, but could you tell me why this happend?

**Comment by BossZou** [comment]
@HuangdaxianZaixian , in my opinion, it has a lot to do with grpc. You may ask for grpc developer if you want understand what happend.
